File: Gestao_Riscos/Codigo/front-gestao-riscos/src/repositories/usuarioRepository.js
import Repository from "./Repository";
const resource = "/usuarios";

export default {
    getByAutenticacao() {
        return Repository.get(`${resource}/usuario`);
    },

    findByUnidade(unidade) {
        return Repository.get(`${resource}?unidade=${unidade}`);
    },
    
}

File: Gestao_Riscos/Codigo/back-gestao-riscos/src/main/java/br/ufc/quixada/npi/gestaoriscos/repository/UsuarioRepository.java
package br.ufc.quixada.npi.gestaoriscos.repository;

import br.ufc.quixada.npi.gestaoriscos.model.Unidade;
import br.ufc.quixada.npi.gestaoriscos.model.Usuario;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface UsuarioRepository extends JpaRepository<Usuario, Integer> {

    Usuario findByEmail(String email);

    @Query("FROM Usuario u WHERE u.unidade IN :unidades ORDER BY u.nome")
    List<Usuario> findByUnidade(List<Unidade> unidades);
}
